### ISSUE TITLE
Analyze performance for exporting data packages

### ISSUE THREAD

**Issue by vitorbaptista** [ckan/ckanext-datapackager#30]
The issue #26 built the code for exporting datapackages but, as there's no caching schema, the exports happen on demand. It'll probably have poor performance, specially on datasets with many/big resources. We should investigate that and, if there's a problem, look for a solution (probably involving caching).

**Comment by pwalsh** [comment]
@amercader any ideas/thoughts on this?

**Comment by amercader** [comment]
> as there's no caching schema, the exports happen on demand.

What is the actual issue here, that the schema needs to be inferred from the actual file and so we need to read it? Otherwise
Are we not storing all extra fields from datapackage.json (including the schema) as dataset/resource extras?

**Comment by vitorbaptista** [comment]
@amercader A user could ask for any dataset to be exported as a data package, not only those who were originally imported from data packages. This means that, at minimum, we need to:

1. Generate the `datapackage.json`;
2. Zip the resource files together.

To avoid these issues, we currently just export a `datapackage.json` linking to the resources in CKAN.

**Comment by amercader** [comment]
OK, 1 I don't think it will be that cost intensive (we are doing that already for all datasets regardless of whether they came from a datapackage.json or not right?) but 2 should be run on a queue.

**Comment by vitorbaptista** [comment]
Yes, `1` is OK for now. However, when we start generating the spec of the resources, we might have to download external files, which can be slow. Unless we somehow use the datastore metadata to generate the descriptor file.

I'd say the entire data package generation should be ran in a queue, with their results cached.

**Comment by amercader** [comment]
> when we start generating the spec of the resources, we might have to download external files
If someone didn't upload the file to CKAN maybe it was for a good reason (eg it's 5 Gb). I would just try to guess the schema of stuff that has been uploaded to CKAN and/or it's on the DataStore.

But +1 on having it all run in a queue. Time to finish off https://github.com/ckan/ckan/pull/2706! ($$$ to be made https://github.com/ckan/ckan/issues/2977 :p )
